Automattic/wp-calypso
# Jetpack Backup: fix Jetpack review prompt Tracks event to trigger only when the review prompt is rendered

Related to https://github.com/Automattic/jetpack-backup-team/issues/489

## Proposed Changes

* Fix the `calypso_jetpack_review_prompt_view` Tracks event to trigger only when the review prompt is rendered

## Why are these changes being made?


Currently, we are triggering the `calypso_jetpack_review_prompt_view` event on every restore even if it is not being rendered, which can cause misleading analytics. The idea is to trigger it only if it is rendered.

We stopped rendering this view when the user already dismissed it by clicking the `X` or `Leave a review`.

![CleanShot 2024-06-04 at 01 20 00@2x](https://github.com/Automattic/wp-calypso/assets/1488641/d4c1ed6a-7c7f-4d9b-a837-26f9ca5129d2)

## Testing Instructions



* Spin up a Jetpack Cloud live branch
* Open the Network tab in your developer tools
* Navigate to Jetpack Cloud > Backup
* Try to run a restore with any backup point
* If you see the Jetpack plugin review prompt, ensure you see a request to `t.gif` with the `calypso_jetpack_review_prompt_view` event.
* If you don't see it, ensure you don't see any request to `t.gif` with the `calypso_jetpack_review_prompt_view`event.

If you see the plugin review prompt, you can click on the `X` or `Leave a review` and then try a new restore.

## Pre-merge Checklist
